var/spack/repos/builtin/packages/sundials/package.py

```diff
@@ -27,138 +27,341 @@
 import sys
 
 
-class Sundials(Package):
+class Sundials(CMakePackage):
     """SUNDIALS (SUite of Nonlinear and DIfferential/ALgebraic equation
     Solvers)"""
 
-    homepage = "http://computation.llnl.gov/casc/sundials/"
-    url = "http://computation.llnl.gov/projects/sundials-suite-nonlinear-differential-algebraic-equation-solvers/download/sundials-2.6.2.tar.gz"
+    homepage = "https://computation.llnl.gov/projects/sundials"
+    url = "https://computation.llnl.gov/projects/sundials/download/sundials-2.7.0.tar.gz"
+    maintainers = ['cswoodward', 'gardner48']
 
-    version('2.7.0', 'c304631b9bc82877d7b0e9f4d4fd94d3')
+    # ==========================================================================
+    # Versions
+    # ==========================================================================
+
+    version('3.0.0-beta-2',
+            git='https://github.com/LLNL/sundials.git', tag='v3.0.0-beta-2')
+    version('2.7.0', 'c304631b9bc82877d7b0e9f4d4fd94d3', preferred=True)
     version('2.6.2', '3deeb0ede9f514184c6bd83ecab77d95')
 
+    # ==========================================================================
+    # Variants
+    # ==========================================================================
+
+    # SUNDIALS solvers
+    sun_solvers = ['CVODE', 'CVODES', 'ARKODE', 'IDA', 'IDAS', 'KINSOL']
+
+    for pkg in sun_solvers:
+        variant(pkg, default=True,
+                description='Enable %s solver' % pkg)
+
+    # Real type
+    variant(
+        'precision',
+        default='double',
+        description='''real type precision''',
+        values=('single', 'double', 'extended'),
+        multi=False
+    )
+
+    # Index type
+    variant(
+        'indextype',
+        default='int64_t',
+        description='''index integer type''',
+        values=('int32_t', 'int64_t'),
+        multi=False
+    )
+
+    # Parallelism
     variant('mpi',     default=True,
-            description='Enable MPI parallelism')
-    variant('lapack',  default=True,
-            description='Use external BLAS/LAPACK libraries')
-    variant('klu',     default=False,
+            description='Enable MPI parallel vector')
+    variant('openmp',  default=False,
+            description='Enable OpenMP parallel vector')
+    variant('pthread', default=False,
+            description='Enable Pthreads parallel vector')
+    variant('cuda',    default=False,
+            description='Enable CUDA parallel vector')
+    variant('raja',    default=False,
+            description='Enable RAJA parallel vector')
+
+    # External libraries
+    variant('lapack',     default=False,
+            description='Enable LAPACK direct solvers')
```

Comment:
so that's something you use in Sundials regardless of `superlu_mt` or alike? So that's indeed a build option?

Reply:
Yes, this activates interfaces to use `LAPACK`dense or banded solvers for linear systems that arise inside `sundials` time integrators and nonlinear solvers.